pageserver/src/tenant/mgr.rs

```diff
@@ -214,15 +237,19 @@ impl TenantsMap {
                             }
 
                             if Some(tenant.shard_identity.number) == want_shard {
-                                return Some(*slot.0);
+                                return ShardResolveResult::Found(tenant.clone());
                             }
                         }
                         _ => continue,
                     }
                 }
 
                 // Fall through: we didn't find an acceptable shard
```

Comment:
This comment is only confusing (me, at least)

Reply:
Made it clearer in b968ae0aa8a6bb5ccd5fabede0c6a4f25a5163cb